### PR TITLE
Fix: utils: Add 'sudo' only when there is a sudoer(bsc#1215549)

### PR DESCRIPTION
**Pull request by liangxin1300**
backport #1247

### CODE REVIEW

**Review by nicholasyang2022 — APPROVED** [review]

**Review by nicholasyang2022 — CHANGES_REQUESTED** [review]
It is preferred to avoid using split to parse cmdline arguments to avoid shell injection.

**Comment by liangxin1300** [comment]
> It is preferred to avoid using split to parse cmdline arguments to avoid shell injection.

Changed on all 3 PRs

Thanks for pointing this!

**Comment by nicholasyang2022** [comment]
This change does not work. If whitespace characters exist in `host`, `shlex.split` will still split it. To avoid shell injection, arguments should not be composed into a command line without validation. For this case, it is preferred to operate on the argument array instead of command line string.

**Comment by liangxin1300** [comment]
> This change does not work. If whitespace characters exist in `host`, `shlex.split` will still split it. To avoid shell injection, arguments should not be composed into a command line without validation. For this case, it is preferred to operate on the argument array instead of command line string.

Changed
Please take a look.
Thanks!

**Review by nicholasyang2022 — APPROVED** [review]
~~{"ssh", "true"} is not the same as {"ssh", "", "true"}. However, it seems that ssh ignore an empty arg and this does not cause any problem.~~

File: crmsh/utils.py [hunk 1]
```diff
@@ -185,7 +185,8 @@ def _get_user_of_host_from_config(host):
     def _guess_user_for_ssh(host: str) -> typing.Tuple[str, str]:
         args = ['ssh']
         args.extend(constants.SSH_OPTION_ARGS)
-        args.extend(['-o', 'BatchMode=yes', host, 'sudo', 'true'])
+        sudo_str = 'sudo ' if userdir.get_sudoer() else ''
+        args.extend(['-o', 'BatchMode=yes', host, sudo_str, 'true'])
```
Comment (nicholasyang2022) [inline crmsh/utils.py:189]: 1. When passing arguments as array of strings, space is not needed. Space is the separator of arguments in shell. See `init main(int argc, char **argv)` and `execve(1)`.
2. An empty argument is not the same as non-existing argument. `argv = {"true"}` and  `argv = {"", "true"}` is not the same.

```suggestion
        if userdir.get_sudoer():
            args.extend(['-o', 'BatchMode=yes', host, 'sudo', 'true'])
        else:
            args.extend(['-o', 'BatchMode=yes', host, 'true'])
```

**Review by nicholasyang2022 — APPROVED** [review]

**Pull request merged**